streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -293,6 +327,8 @@
      *                          if not specified the default serdes defined in the configs will be used
      * @param <V1>              the value type of the other stream
      * @param <R>               the value type of the new stream
+     *
+     * @return a {@link KStream} that contains join-records for each key and values computed by the given {@link ValueJoiner}
```

Comment:
Nit: we sometimes use "that contains", and sometimes use "containing". Either should be fine, we can just make it consistent.

Also: "joined records computed by the given {@link ValueJoiner}, one for each matched record-pairs with the same key and within the joining window intervals." Similar below.